js/ui/grid_core/ui.grid_core.virtual_scrolling.js

```diff
@@ -758,6 +752,10 @@ export const virtualScrollingModule = {
             data: (function() {
                 const members = {
                     _refreshDataSource: function() {
+                        const shouldEnableVirtualRowRenderingMode = this.option(NEW_SCROLLING_MODE) && (isVirtualMode(this) || isAppendMode(this)) && !isVirtualRowRendering(this);
+                        if(shouldEnableVirtualRowRenderingMode) {
+                            this._silentOption(ROW_RENDERING_MODE_OPTION, SCROLLING_MODE_VIRTUAL);
```

Comment:
Replace _silentOption call to using utility method in all places where rowRenderingMode option uses.